MetaMorpheus/EngineLayer/Gptmd/GptmdEngine.cs

```diff
@@ -60,134 +62,124 @@ public static bool ModFits(Modification attemptToLocalize, IBioPolymer protein,
-                                if (possibleIndices.Any())
+                                foreach (var mod in possibleModifications)
                                 {
-                                    List<PeptideWithSetModifications> newPeptides = new();
-                                    foreach (int index in possibleIndices)
+                                    List<int> possibleIndices = Enumerable.Range(0, pepWithSetMods.Length).Where(i => ModFits(mod, pepWithSetMods.Parent, i + 1, pepWithSetMods.Length, pepWithSetMods.OneBasedStartResidue + i)).ToList();
+                                    if (possibleIndices.Any())
                                     {
-                                        if (mod.MonoisotopicMass.HasValue)
+                                        List<PeptideWithSetModifications> newPeptides = new();
+                                        foreach (int index in possibleIndices)
                                         {
-                                            newPeptides.Add((PeptideWithSetModifications)pepWithSetMods.Localize(index, mod.MonoisotopicMass.Value));
+                                            if (mod.MonoisotopicMass.HasValue)
+                                            {
+                                                newPeptides.Add((PeptideWithSetModifications)pepWithSetMods.Localize(index, mod.MonoisotopicMass.Value));
+                                            }
                                         }
-                                    }
 
-                                    if (newPeptides.Any())
-                                    {
-                                        var scores = new List<double>();
-                                        var dissociationType = CommonParameters.DissociationType == DissociationType.Autodetect ?
-                                            psms[i].MsDataScan.DissociationType.Value : CommonParameters.DissociationType;
+                                        if (newPeptides.Any())
+                                        {
+                                            var scores = new List<double>();
+                                            var dissociationType = CommonParameters.DissociationType == DissociationType.Autodetect ?
+                                                psms[i].MsDataScan.DissociationType.Value : CommonParameters.DissociationType;
 
-                                        scores = CalculatePeptideScores(newPeptides, dissociationType, psms[i]);
+                                            scores = CalculatePeptideScores(newPeptides, dissociationType, psms[i]);
 
-                                        // If the score is within tolerance of the highest score, add the mod to the peptide
-                                        // If the tolerance is too tight, then the number of identifications in subsequent searches will be reduced
-                                        double scoreTolerance = 0.1;
```

Comment:
Magic number should probably be a property

Reply:
done